### PR TITLE
Filter images by reference

### PR DESCRIPTION
**Pull request by manugupt1**
Adds support for filtering images by reference.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -156,4 +156,4 @@ jobs:
         run: |
           sudo apt-get install -y expect
       - name: "Ensure that the integration test suite is compatible with Docker"
-        run: go test -v -exec sudo ./cmd/nerdctl/... -args -test.target=docker -test.kill-daemon
+        run: go test -timeout 30m -v -exec sudo ./cmd/nerdctl/... -args -test.target=docker -test.kill-daemon
```
Comment (Zheaoli) [inline .github/workflows/test.yml:159]: Why do we need this?

**Review by Zheaoli — COMMENTED** [review]
LGTM for the code part

File: cmd/nerdctl/images.go [hunk 2]
```diff
@@ -144,6 +148,30 @@ func imagesAction(cmd *cobra.Command, args []string) error {
...
 
+func filterByReference(imageList []images.Image, filters []string) ([]images.Image, error) {
+	var filteredImageList []images.Image
+	logrus.Debug(filters)
+	for _, image := range imageList {
+		logrus.Debug(image.Name)
+		var matches int
+		for _, f := range filters {
+			var err error
+			matchFound, err := regexp.MatchString(f, image.Name)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images.go:159]: Looks like it is expected to support wildcard, not regexp?
https://docs.docker.com/engine/reference/commandline/images/#filter-images-by-reference

```
docker images --filter=reference='busy*:*libc'
```

Comment (AkihiroSuda) [reply cmd/nerdctl/images.go:159]: We should also make sure that `--filter=reference='busy*:*libc'` matches `docker.io/library/busy*:*libc`

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**